features/workflow.feature

```diff
@@ -16,125 +16,122 @@ Feature: Workflow
         - Workflows must be one-directional (no loops)
         - All subsequent jobs must checkout from the same SHA as the previous job
 
-    Scenario: Failure
-        Given an existing pipeline with the workflow:
-            | job | requires |
-            | FOO | ~commit  |
-            | BAR | FOO      |
-        When the "FOO" job is started
-        And the build failed
-        Then the "BAR" job is not started
-
     Scenario: Serially
-        Given an existing pipeline with the workflow:
-            | job | requires |
-            | FOO | ~commit  |
-            | BAR | FOO      |
-        When the "FOO" job is started
-        And the build succeeded
-        Then the "BAR" job is started
-        And that "BAR" build uses the same SHA as the "FOO" build
+        Given an existing pipeline on "serially" branch with the workflow jobs:
+            | job       | requires  |
+            | SIMPLE    | ~commit   |
+            | PARALLEL1 | SIMPLE    |
+
+    Scenario: Failure
+        Given an existing pipeline on "failure" branch with the workflow jobs:
+            | job           | requires  |
+            | FAIL          | ~commit   |
+            | AFTER-FAIL    | FAIL      |
+        When a new commit is pushed to "failure" branch
+        When the "FAIL" job is triggered
+        And the "FAIL" build failed
+        Then the "AFTER-FAIL" job is not triggered
 
     Scenario: Parallel
-        Given an existing pipeline with the workflow:
-            | job | requires |
-            | FOO | ~commit  |
-            | BAR | FOO      |
-            | BAZ | FOO      |
-        When the "FOO" job is started
-        And the build succeeded
-        Then the "BAR" job is started
-        And the "BAZ" job is started
-        And that "BAR" build uses the same SHA as the "FOO" build
-        And that "BAZ" build uses the same SHA as the "FOO" build
+        Given an existing pipeline on "parallel" branch with the workflow jobs:
+            | job       | requires  |
+            | SIMPLE    | ~commit   |
+            | PARALLEL1 | SIMPLE    |
+            | PARALLEL2 | SIMPLE    |
+        When a new commit is pushed to "parallel" branch
+        And the "SIMPLE" job is triggered
+        And the "SIMPLE" build succeeded
+        Then the "PARALLEL1" job is triggered from "SIMPLE"
+        And the "PARALLEL2" job is triggered from "SIMPLE"
+        And that "PARALLEL1" build uses the same SHA as the "SIMPLE" build
+        And that "PARALLEL2" build uses the same SHA as the "SIMPLE" build
+        And the "STAGING" job is not triggered
```

Comment:
Why need bottom 2 lines from here?

Reply:
We don't need these lines, I will remove.